Content.Server/Silicons/Bots/MedibotSystem.cs

```diff
@@ -0,0 +1,175 @@
+using System.Diagnostics.CodeAnalysis;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Damage;
+using Content.Shared.DoAfter;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Popups;
+using Content.Shared.ActionBlocker;
+using Content.Shared.Interaction;
+using Content.Shared.FixedPoint;
+using Content.Server.DoAfter;
+using Content.Server.Popups;
+using Content.Server.NPC.Components;
+using Content.Server.Chemistry.EntitySystems;
+using Robust.Shared.Physics.Components;
+using Robust.Server.GameObjects;
+
+namespace Content.Server.Silicons.Bots
+{
+    public sealed class MedibotSystem : EntitySystem
+    {
+        [Dependency] private readonly MobStateSystem _mobs = default!;
+        [Dependency] private readonly SolutionContainerSystem _solution = default!;
+        [Dependency] private readonly PopupSystem _popups = default!;
+        [Dependency] private readonly DoAfterSystem _doAfter = default!;
+        [Dependency] private readonly AudioSystem _audioSystem = default!;
+        [Dependency] private readonly ActionBlockerSystem _blocker = default!;
+        [Dependency] private readonly PhysicsSystem _physics = default!;
+        [Dependency] private readonly SharedInteractionSystem _interactionSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<MedibotComponent, InteractNoHandEvent>(PlayerInject);
```

Comment:
This could just call NPCStartInject (now called StartInject after this fix) in a lambda that checks the target for validity and passes it in.